src/components/Modals/SemesterMenu.vue

```diff
@@ -19,6 +19,25 @@
         </div>
       </div>
     </button>
+    <button
+      v-if="!isOpenModal"
+      class="semesterMenu-section full-opacity-on-hover"
+      @click="openClearSemesterModal"
+      data-cyId="semesterMenu-clear"
+    >
+      <div class="semesterMenu-content">
+        <div class="semesterMenu-content">
+          <div class="semesterMenu-left">
+            <img
+              class="semesterMenu-icon"
+              src="@/assets/images/erase.svg"
+              alt="clear semester erase icon"
+            />
+            <span class="semesterMenu-edit">Clear Semester</span>
+          </div>
+        </div>
+      </div>
+    </button>
```

Comment:
Do you think it would be possible to factor out some of the repeated code for each modal into a separate component? The same button template seems to be similar for each of the edit, clear, and delete modals.